crates/broker/src/lib.rs

```diff
@@ -217,14 +244,34 @@ impl Order {
             expire_timestamp: None,
             client_sig,
             lock_price: None,
+            fulfillment_type,
             error_msg: None,
         }
     }
+
+    // An Order is identified by the request_id, the fulfillment type, and the hash of the proof request.
+    // This structure supports multiple different ProofRequests with the same request_id, and different
+    // fulfillment types.
+    pub fn id(&self) -> String {
+        format!(
+            "0x{:x}-{}-{:?}",
+            self.request.id,
+            self.request.signing_hash(Address::ZERO, 1).unwrap(),
```

Comment:
why is this hash necessary?

Reply:
This is for future support of multiple requests with the same ID https://github.com/boundless-xyz/boundless/issues/162

Reply:
Thats right, it has been useful to have it in the id for debugging. 

Good callout on the chain and contract- I meant to come back to this actually but forgot!

And correct on fulfillment type. I think having it operate separately keeps things a lot cleaner and easier to debug, and reduces risks of various race conditions.

Regarding proof work in progress, I think we should add a caching layer in the prover to the executor runs and proof runs (I think this is something we should do regardless of this PR anyway for DDOS/spam protection)